pkg/cmd/admin/router/router.go

```diff
@@ -680,11 +680,48 @@ func generateStatsPassword() string {
 	return strings.Join(password, "")
 }
 
-func validateServiceAccount(kClient *kclient.Client, ns string, sa string) error {
+// associate the secrets with the account given (within the namespace)
+func associateSecrets(kClient *kclient.Client, secrets []*kapi.Secret, ns string,
+	sa string, objects []runtime.Object) ([]runtime.Object, error) {
+	if len(secrets) != 0 {
```

Comment:
minor nit: `len(secrets) > 0` seems to parse better with my brain rather than the not equal to - aka do it only if there are any secrets (and lies!!).